crates/primitives/src/proofs.rs

```diff
@@ -92,19 +85,15 @@ where
     })
 }
 
-/// Calculates the log root for headers.
-pub fn calculate_log_root(logs: &[Log]) -> B256 {
-    //https://github.com/ethereum/go-ethereum/blob/356bbe343a30789e77bb38f25983c8f2f2bfbb47/cmd/evm/internal/t8ntool/execution.go#L255
-    let mut logs_rlp = Vec::new();
-    alloy_rlp::encode_list(logs, &mut logs_rlp);
-    keccak256(logs_rlp)
-}
-
 /// Calculates the root hash for ommer/uncle headers.
 pub fn calculate_ommers_root(ommers: &[Header]) -> B256 {
     // RLP Encode
     let mut ommers_rlp = Vec::new();
     alloy_rlp::encode_list(ommers, &mut ommers_rlp);
+    // Check if `ommers` list is empty
+    if ommers_rlp.is_empty() {
+        return EMPTY_ROOT_HASH
+    }
```

Comment:
this can be moved to the beginning of the function, so we don't even allocate an rlp header

Reply:
@mattsse Correct, I misnamed the hash on my first try so it didn't work, now it's ok. Just pushed the update.